src/v/cloud_storage/partition_recovery_manager.cc

```diff
@@ -191,9 +191,7 @@ ss::future<log_recovery_result> partition_downloader::download_log() {
     }
     vlog(_ctxlog.info, "Downloading log for {}", _ntpc.ntp());
     try {
-        auto topic_manifest_path = topic_manifest::get_topic_manifest_path(
-          _ntpc.ntp().ns, _ntpc.ntp().tp.topic);
-        co_return co_await download_log(topic_manifest_path);
+        co_return co_await download_log_inner();
```

Comment:
nit: maybe rename `download_log()` to `maybe_download_log()` and rename `download_log_inner()` to `download_log()` or `download_manifest_and_segments()` or somesuch?